### ISSUE TITLE
SP6 power-profiles-daemon can't switch to power saver mode

### ISSUE THREAD

**Issue by xan105** [linux-surface/linux-surface#795]
Whenever I try to switch to the power saver mode it resets back to balanced.
I can switch to performance, balanced but not power saver.

Using the gnome tray, gnome settings, or powerprofilesctl yield no difference.

I saw on the power-profiles-daemon gitlab that I can debut it with
`sudo G_MESSAGES_DEBUG=all /usr/libexec/power-profiles-daemon -r -v`

<details><summary>power-profiles-daemon output</summary>
<pre>
** (power-profiles-daemon:8619): DEBUG: 00:31:01.097: Starting power-profiles-daemon version 0.10.1
** (power-profiles-daemon:8619): DEBUG: 00:31:01.101: Handling driver 'fake'
** (power-profiles-daemon:8619): DEBUG: 00:31:01.101: probe() failed for driver fake, skipping
** (power-profiles-daemon:8619): DEBUG: 00:31:01.101: Handling driver 'platform_profile'
(power-profiles-daemon:8619): GLib-GIO-DEBUG: 00:31:01.103: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: ACPI performance_profile is now b, so profile is detected as balanced
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Didn't find a dytc_lapmode sysfs attribute to thinkpad_acpi
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Handling driver 'intel_pstate'
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Driver 'platform_profile' already probed, skipping driver 'intel_pstate'
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Handling driver 'placeholder'
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Driver 'platform_profile' already probed, skipping driver 'placeholder'
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Handling action 'trickle_charge'
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Applying profile 'balanced' from configuration file
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Setting active profile 'balanced' for reason 'reset' (current: 'balanced')
** (power-profiles-daemon:8619): DEBUG: 00:31:01.114: Can't switch to balanced mode, already there
** (power-profiles-daemon:8619): DEBUG: 00:31:29.953: Transitioning active profile from 'balanced' to 'performance' by user request
** (power-profiles-daemon:8619): DEBUG: 00:31:29.953: Setting active profile 'performance' for reason 'user' (current: 'balanced')
** (power-profiles-daemon:8619): DEBUG: 00:31:29.955: Successfully switched to profile performance
** (power-profiles-daemon:8619): DEBUG: 00:31:29.980: /sys/firmware/acpi/platform_profile changed
** (power-profiles-daemon:8619): DEBUG: 00:31:29.981: ACPI performance_profile is now p, so profile is detected as performance
** (power-profiles-daemon:8619): DEBUG: 00:31:29.981: /sys/firmware/acpi/platform_profile changed
** (power-profiles-daemon:8619): DEBUG: 00:31:29.983: ACPI performance_profile is now p, so profile is detected as performance
** (power-profiles-daemon:8619): DEBUG: 00:31:29.983: /sys/firmware/acpi/platform_profile changed
** (power-profiles-daemon:8619): DEBUG: 00:31:29.985: ACPI performance_profile is now p, so profile is detected as performance
** (power-profiles-daemon:8619): DEBUG: 00:31:39.815: Transitioning active profile from 'performance' to 'power-saver' by user request
** (power-profiles-daemon:8619): DEBUG: 00:31:39.815: Setting active profile 'power-saver' for reason 'user' (current: 'performance')
** (power-profiles-daemon:8619): DEBUG: 00:31:39.817: Successfully switched to profile power-saver
** (power-profiles-daemon:8619): DEBUG: 00:31:39.841: /sys/firmware/acpi/platform_profile changed
** (power-profiles-daemon:8619): DEBUG: 00:31:39.843: ACPI performance_profile is now b, so profile is detected as balanced
** (power-profiles-daemon:8619): DEBUG: 00:31:39.843: Driver 'platform_profile' switched internally to profile 'balanced' (current: 'power-saver')
** (power-profiles-daemon:8619): DEBUG: 00:31:39.843: Setting active profile 'balanced' for reason 'internal' (current: 'power-saver')
** (power-profiles-daemon:8619): DEBUG: 00:31:39.843: Can't switch to balanced mode, already there
** (power-profiles-daemon:8619): DEBUG: 00:31:39.854: /sys/firmware/acpi/platform_profile changed
** (power-profiles-daemon:8619): DEBUG: 00:31:39.855: ACPI performance_profile is now b, so profile is detected as balanced
** (power-profiles-daemon:8619): DEBUG: 00:31:39.856: /sys/firmware/acpi/platform_profile changed
** (power-profiles-daemon:8619): DEBUG: 00:31:39.857: ACPI performance_profile is now b, so profile is detected as balanced
</pre>

</details>

There doesn't seem to be any error but I don't understand/know why it says "Successfully switched to profile power-saver"
and yet driver 'platform_profile' forced it to balanced.

I tried to search similar issues but they seemed resolved/closed.
I don't know how to proceed further to debug/help with this.
I don't think this is normal behavior ? 

### Environment

- Hardware model: Surface Pro 6
- Kernel version: 5.16.17-1.surface.fc35.x86_64
- Distribution: Fedora 35
- DE: Gnome 41.5

`dmesg` output: [dmesg.txt](https://github.com/linux-surface/linux-surface/files/8524113/dmesg.txt)

**Comment by sparkie3** [comment]
I'm having this exact same issue on a SP6. I'm using Plasma with a default fedora 36 installed, and Plasma installed on a separate user. The issue persists on Gnome and Plasma, and from the command line.

It seems like something somewhere is changing it back to balanced.

**Comment by qzed** [comment]
As discussed on matrix: This is likely a firmware problem and it looks like we can't do anything about it from the kernel. Essentially, the firmware refuses to accept the low-power mode.

**Comment by sparkie3** [comment]
Yea, that was me walking through it. I wonder what changed, because it worked fine from Feb until a couple months ago., then just stopped working. I didn't have windows on it at all, so unless fwupd automatically updates the firmware, something changed.

I don't know if a specific time frame can help narrow it down.

Is it possible that some other driver is over-riding the power mode?

**Comment by NP-chaonay** [comment]
could you all have this problem affected, try to unplug and a few seconds (5sec maybe enough) try to reswitching.

But if I not remember wrong, I have seem have remember that there is another github issue like this but even on battery, no low power mode can be activated

**Comment by qzed** [comment]
I doubt that fwudp updated something automatically. Unfortunately that's a black box, I wish we had some more insights into what's going on inside that.

**Comment by sparkie3** [comment]
> 

Perhaps if someone with windows installed can get the commands sent so the two can be compared.

**Comment by qzed** [comment]
Right, I think the best option is checking what Windows does here (and whether that happens there, too) with IRPMon.

**Comment by sparkie3** [comment]
@qzed So at the moment on fedora 41 switching to power saver works. I need to figure out how to check some of the clocks and the power usage, but I'll see if it does in fact change the profile or  if it just no longer shows the rejection in the gnome applet.

I tried for along time to make KDE Plasma work on the surface but it's just not very good. KDE Mobile loos very, very promising.
